docs/deposit.md

```diff
@@ -0,0 +1,168 @@
+  	pool.Status = types.PoolStatus_POOL_STATUS_READY
+  }
+
+    newDepositID := len(depositLedger[msg.sender])
+    newDeposit := DepositMetadata {
+        ID: newDepositID,
+        PoolToken: poolToken,
+        ExpireDate: ctx.BlockTime(),
+    }
+
+    k.AddNewDepositToLedger(ctx, msg.sender, newDeposit)
+  ```
+
+  - update `OnSingleDepositAcknowledged`
+
+  ```
+  func (k Keeper) OnSingleDepositAcknowledged(ctx sdk.Context, req *types.MsgDepositRequest, res *types.MsgDepositResponse) error {
+    ...
+    newDepositID := len(depositLedger[req.sender])
+    newDeposit := DepositMetadata {
+        ID: newDepositID,
+        PoolToken: *res.PoolToken,
+        ExpireDate ctx.BlockTime() + msg.Tick,
+    }
+    k.AddNewDepositToLedger(ctx, msg.sender, newDeposit)
+  }
+  ```
+
+  - update withdraw
+
+  ```
+   func (k Keeper) OndWithdrawReceive(ctx sdk.Context, msg *types.MsgWithdrawRequest) (*types.MsgWithdrawResponse, error) {
+
+  	if err := msg.ValidateBasic(); err != nil {
+  		return nil, err
+  	}
+  	pool, found := k.GetInterchainLiquidityPool(ctx, msg.PoolCoin.Denom)
+
+  	if !found {
+  		return nil, types.ErrNotFoundPool
+  	}
+
+      depositLedger := GetDepositLedger(ctx, msg.sender)
+      withdrawableAmount := 0
+      startIndex := 0
+      for index, metadata := range depositLedger {
+          if metadata.ExpireDate > time.now() {
+              withdrawableAmount.add(metadata.PoolToken)
+          }else{
+              break
+          }
+          if withdrawableAmount >=  msg.Tokens[0] {
+              startIndex = index
+              remainAmount := withdrawableAmount.sub(metadata.PoolToken)
+              if remainAmount.Uint64() > sdk.NewInt(0) {
+                  depositLedger[index].Tokens = remainAmount
+                   startIndex = index
+              }else{
+                  startIndex = index+1
+              }
+              break
+          }
+      }
+
+    if withdrawableAmount < msg.Token[0] {
+        return nil, ErrNotEnoughWithdrawableTokens
+        }
+
+      if startIndex != 0 {
+          updatedDepositLedger = depositLedger[startIndex:len(depositLedger)]
+          k.SetDepositLedger(ctx, msg.sender, updatedDepositLedger)
+      }
+      ...
+  }
+  ```
+
+Add the same update logic `OnWithdrawAcknowledged`.
+
+##### Note
+
+- In this case, users obtain their LP tokens immediately within their wallets.
+- However, there is a loop to calculate the withdrawable amount. If the ledger continues to grow, it may impact the chain's performance. Implementing a limitation on the number of deposits may be a better approach. We can assume that a user is a malicious actor if they do not withdraw at all and only attempt to deposit multiple times. Additionally, we should establish a minimum deposit amount to prevent attacks.
+
+#### Plan 2
+
+##### Scenario
+
+In this method, we do not directly provide Pool Tokens to users. Instead, we supply an NFT that includes specific metadata.
```

Comment:
It's interesting, but it will introduce a lot of things related to NFT. It would be better to keep the protocol as simple as possible.